src/ast/passes/semantic_analyser.cpp

```diff
@@ -1529,30 +1529,77 @@ Probe *SemanticAnalyser::get_probe_from_scope(Scope *scope,
 
 void SemanticAnalyser::visit(Map &map)
 {
-  MapKey key;
+  SizedType key = CreateNone();
+  bool is_tuple_ast = false;
+
+  if (map.key_expr) {
+    Visit(map.key_expr);
+    Expression *expr = map.key_expr;
+
+    if (expr->type.IsTupleTy()) {
+      // We can only do this expression and size type manipulation
+      // if we have access to the underlying expressions in the tuple ast
+      // e.g. @map[(1, 2)] vs $a = (1, 2) @map[$a]
```

Comment:
Is this definitely true? We have `expr->type` which holds the type information of the tuple, no matter whether it's a literal tuple or a variable / map holding a tuple. I'd have thought we'd be able to get the type information out of that without looking at the AST directly.

This'd let us get rid of that incompatible string-length limitation when using non-literal tuples.